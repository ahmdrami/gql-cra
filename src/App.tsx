import React from 'react'
import { gql, useQuery } from '@apollo/client'

export const ME_QUERY = gql`
  query {
    me
  }
`
type Data = {
  me: number
}
function App() {
  const { loading, error, data } = useQuery<Data>(ME_QUERY)

  if (loading) return <p>Loading...</p>
  if (error) return <p>Error :(</p>

  return <div className="App">User: {data?.me}</div>
}

export default App
