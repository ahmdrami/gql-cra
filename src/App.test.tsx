import React from 'react'
import { MockedProvider } from '@apollo/client/testing'
import { render, screen, waitFor } from '@testing-library/react'
import App, { ME_QUERY } from './App'

describe('App component', () => {
  test('should render the fragment', async () => {
    const mocks = [
      {
        request: {
          query: ME_QUERY,
        },
        result: {
          data: {
            me: 12314123,
          },
        },
      },
    ]
    const { asFragment } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>
    )

    await waitFor(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0))
    })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render and verify me user in the document', async () => {
    const mocks = [
      {
        request: {
          query: ME_QUERY,
        },
        result: {
          data: {
            me: 12314123,
          },
        },
      },
    ]
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>
    )

    await waitFor(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0))
    })

    const linkElement = screen.getByText(/12314123/)
    expect(linkElement).toBeInTheDocument()
  })
})
