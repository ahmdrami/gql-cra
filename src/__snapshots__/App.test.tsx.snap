// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render the fragment 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    User: 12314123
  </div>
</DocumentFragment>
`;
